File: checkout-ruby/lib/rule_discount_value.rb
class RuleDiscountValue

  attr_reader :rule_discount_value_type, :discount_value

  def initialize(rule_discount_value_type, discount_value)
    @discount_value = discount_value
    @rule_discount_value_type = rule_discount_value_type
  end

  def get_discounted_price(price)
    discount = 0.0
    if rule_discount_value_type == RuleDiscountValueType::PERCENTAGE
      discount = (price * discount_value)/100.00
    elsif rule_discount_value_type == RuleDiscountValueType::ABSOLUTE
      discount = discount_value
    end
    discount.floor(2)
  end


end

File: src/main/java/com/vinterior/checkout/RuleDiscountValue.java
package com.vinterior.checkout;

import com.vinterior.checkout.enums.RuleDiscountValueType;
import lombok.AllArgsConstructor;
import lombok.Getter;

import java.math.RoundingMode;
import java.text.DecimalFormat;

@Getter
@AllArgsConstructor
public class RuleDiscountValue {

   private RuleDiscountValueType ruleDiscountValueType;

   private double ruleDiscountValue;

   public double getDiscountedPrice(double price) {
      double discount = 0.0;
      DecimalFormat df = new DecimalFormat("0.00");
      df.setRoundingMode(RoundingMode.FLOOR);
      if(ruleDiscountValueType.equals(RuleDiscountValueType.PERCENTAGE)) {
         discount = (price * ruleDiscountValue)/100;
      } else if(ruleDiscountValueType.equals(RuleDiscountValueType.ABSOLUTE)) {
         discount = ruleDiscountValue;
      }
      return Double.parseDouble(df.format(discount));
   }
}
